implementation-details.md

```diff
@@ -273,9 +297,14 @@ The `mask_bits` param is calculated as:
 ```
 mask_bits = max(log2(num_items / max_items), 0.0)
 ```
-where `num_items` is the number of current items we want to insert into the filter and `max_items` is the maximum number of items that can be inserted. 
+where `num_items` is the number of current items we want to insert into the filter: `CrdsValue`s, purged values and failed inserts. `max_items` is the maximum number of items that can be inserted. 
 
-The `CrdsFilter`s are partitioned in pull requests between peers selected randomly using their weights calculated as follows: 
+Node gathers a list of known peers from its `crds` and filters out:
+* not active ones (no message received from that peer in the last 60 seconds)
+* peers with different shred version or shred version != 0
+* peers with invalid gossip address
+
+The `CrdsFilter`s are then partitioned in pull requests between the gathered peers selected randomly using their weights calculated as follows: 
```

Comment:
if i remember correctly, the crds filters themsevles aren't partitioned. but the set of `CrdsFilter`s is partitioned. So one CrdsFilter goes to one node and another to another node and so on